permits/printpermit.py

```diff
@@ -0,0 +1,121 @@
+                  "base64," + base64.b64encode(response.content).decode("utf-8"))
+
+    return map_base64
+
+
+def printreport(request, permit_request_id):
+
+    permit_request = models.PermitRequest.objects.get(pk=permit_request_id)
+
+    geo_times = models.PermitRequestGeoTime.objects.filter(permit_request=permit_request).all()
+    map_image = get_map_base64(geo_times, permit_request.pk)
+
+    print_date = datetime.datetime.now()
+    administrative_entity = gpf_models.AdministrativeEntity.objects.get(
+        pk=permit_request.administrative_entity.pk)
+
+    properties_form = forms.WorksObjectsPropertiesForm(instance=permit_request)
+    properties_by_object_type = dict(properties_form.get_fields_by_object_type())
+    appendices_form = forms.WorksObjectsAppendicesForm(instance=permit_request)
+    appendices_by_object_type = dict(appendices_form.get_fields_by_object_type())
+
+    validations = permit_request.validations.all()
+
+    objects_infos = [
+        (
+            obj,
+            properties_by_object_type.get(obj, []),
+            appendices_by_object_type.get(obj, [])
+        )
+        for obj in permit_request.works_object_types.all()
+    ]
+
+    actor_types = dict(models.ACTOR_TYPE_CHOICES)
+
+    contacts = [
+        (actor_types.get(contact['actor_type'].value(), ''), [
+            (field.label, field.value())
+            for field in contact
+            if field.name not in {'id', 'actor_type'}
+        ])
+        for contact in services.get_permitactorformset_initiated(permit_request)
+        if contact['id'].value()
+    ]
+
+    author = gpf_models.Actor.objects.get(pk=permit_request.author.pk)
+    author_form = gpf_forms.GenericActorForm(instance=author)
+
+    html = render(request, "permits/print/printpermit.html", {
+        'permit_request': permit_request,
+        'contacts': contacts,
+        'author': author_form,
+        'objects_infos': objects_infos,
+        'print_date': print_date,
+        'administrative_entity': administrative_entity,
+        'geo_times': geo_times,
+        'map_image': map_image,
+        'validations': validations,
+        'logo_main': base64.b64encode(administrative_entity.logo_main.open().read()).decode("utf-8") if administrative_entity.logo_main else '',
+        'logo_secondary': base64.b64encode(administrative_entity.logo_secondary.open().read()).decode("utf-8") if administrative_entity.logo_secondary else '',
+        'image_signature_1': base64.b64encode(administrative_entity.image_signature_1.open().read()).decode("utf-8") if administrative_entity.image_signature_1 else '',
+        'image_signature_2': base64.b64encode(administrative_entity.image_signature_2.open().read()).decode("utf-8") if administrative_entity.image_signature_2 else '',
+    })
+
+    pdf_permit = HTML(string=html.content,  base_url=request.build_absolute_uri()).write_pdf(
+        stylesheets=[CSS('/code/static/css/printpermit.css')])  # FIX THAT
```

Comment:
Fix that ;)

Reply:
Actually, ok in docker context but will fail elsewhere. But who's not using docker ? ;-)